### ISSUE TITLE
[FR] 开启F-Droid可重复构建

### ISSUE THREAD

**Issue by hushenghao** [LibChecker/LibChecker#907]
### Checklist / 检查清单

- [X] No one has submitted a similar or identical feature request before. / 之前没有人提交过类似或相同的功能请求。
- [X] This suggestion does not depart from the original intention of LibChecker. / 这个建议不会背离 LibChecker 的初衷。

### Enhancement propose / 改进目的

除了Play商店以外，可以让应用商店统一签名，避免无法跨平台更新的问题。
目前F-Droid平台使用的是F-Droid的签名，开启可重复构建最简单的方案是，F-Droid在构建时使用我们提供的apk地址来使用，例如可以使用github release的apk

### Solution / 解决方案

更新[fdroiddata](https://gitlab.com/fdroid/fdroiddata)仓库中项目的元数据信息，新增如下信息
* [Binaries](https://f-droid.org/zh_Hans/docs/Build_Metadata_Reference/#Binaries): 开发者签名的上游 APK文件地址（github release下载地址）
* [AllowedAPKSigningKeys](https://f-droid.org/zh_Hans/docs/Build_Metadata_Reference/#AllowedAPKSigningKeys): 允许的APK预期签名密钥（github action 中配置的密钥）

参考文档 
* [构建元数据参考](https://f-droid.org/zh_Hans/docs/Build_Metadata_Reference/)
* [F-Droid可重复构建](https://f-droid.org/zh_Hans/docs/Reproducible_Builds/)

### Additional info / 额外信息

目前Releases APK的文件命名规则需要修改，例如：LibChecker-2.4.2.**a14a5ec7**-1755-foss-release.apk，中间的hash值可能导致Binaries字段无法配置
统一 market、foss 和 github action 的apk签名，目前应该是统一的

**Comment by qhy040404** [comment]
@linsui Hello, can you check this?
https://gitlab.com/fdroid/fdroiddata/-/merge_requests/14075

**Comment by linsui** [comment]
As @Licaon-Kter said in the MR, this means the signature is changed from the F-Droid one to the upstream one. The user can't update it. If @zhaobozhen you thought it's OK, please notify the users about this. And someone need to update the F-Droid build manully.

**Comment by qhy040404** [comment]
Does the document mean that every update must upload a signature in the metadata library? or some other solution. @linsui

**Comment by zhaobozhen** [comment]
> As @Licaon-Kter said in the MR, this means the signature is changed from the F-Droid one to the upstream one. The user can't update it. If @zhaobozhen you thought it's OK, please notify the users about this. And someone need to update the F-Droid build manully.

We saw that fdroid provides a way publishing two apks every time(fdroid signed & developer signed). We prefer this way but don't know how to do it. Can you help us with more information?

**Comment by linsui** [comment]
> We prefer this way but don't know how to do it. Can you help us with more information?

Then we need to update the metadata manually as said in https://f-droid.org/zh_Hans/docs/Reproducible_Builds/#%E5%90%8C%E6%97%B6%E5%8F%91%E5%B8%83%E4%B8%8A%E6%B8%B8%E5%BC%80%E5%8F%91%E8%80%85%E7%AD%BE%E5%90%8D%E5%92%8C-f-droid-%E7%AD%BE%E5%90%8D%E7%9A%84-apk. See https://gitlab.com/fdroid/fdroiddata/-/merge_requests/13731 for example.

**Comment by qhy040404** [comment]
If we migrate to publish both signed apks, we need to upload signature every version, and auto update will be disabled because there is no develop signature for the version. Is that right?

**Comment by linsui** [comment]
Yes, we need to disable autoupdate and update the metadata manually.
